fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -94,6 +94,13 @@ class FullyShardedDataParallel(nn.Module):
         since FSDP will shard parameters in-place and this will break any
         previously initialized optimizers.
 
+    .. warning::
+
+        If you wrap every parameter inside a nested FSDP and leaving the outer
+        FSDP empty without any parameter, checkpointing activation may trigger
+        an assert on the backward pass. The solution is to leave some parameters
+        to the outer FSDP.
```

Comment:
oh nice! This was the issue you ran into previously right? I wonder if we can hack this with a zero-element param on the outer-most FSDP instance: `self.params = [Parameter(torch.empty(0))]` or something like that

Reply:
I was thinking about it but I don't know how to hook it into the user's compute graph though. Perhaps keep that as a separate PR when we have an idea? I will add a todo.